azurelinuxagent/ga/exthandlers.py

```diff
@@ -1317,19 +1332,19 @@ def update_settings(self):
 
     def create_handler_env(self):
         handler_env = {
-                "logFolder": self.get_log_dir(),
-                "configFolder": self.get_conf_dir(),
-                "statusFolder": self.get_status_dir(),
-                "heartbeatFile": self.get_heartbeat_file()
+                HandlerEnvironment.logFolder: self.get_log_dir(),
+                HandlerEnvironment.configFolder: self.get_conf_dir(),
+                HandlerEnvironment.statusFolder: self.get_status_dir(),
+                HandlerEnvironment.heartbeatFile: self.get_heartbeat_file()
             }
 
         if is_extension_telemetry_pipeline_enabled():
```

Comment:
This seems to return value of _ENABLE_EXTENSION_TELEMETRY_PIPELINE, which needs a new deployment if changed, correct ? Is it possible to dynamically change the config so that it doesn't require deployment ? (more flexibility). Just thinking

Reply:
It was meant to be that specific and not that dynamic :)
Basically I can also add the flag to the agent config file but we didnt want to give the customers the option to turn this off as this is a very basic workflow needed by the extensions. If we do find an issue, we will turn it off from our end rather and do a successive deployment. That was the whole idea behind it.

Do you have any other ideas by any chance for making it more dynamic but at the same time not opening up the flag to the customers?